hazelcast/src/main/java/com/hazelcast/internal/memory/impl/DirectMemoryBits.java

```diff
@@ -393,53 +553,79 @@ public static char readCharVolatile(long address, boolean bigEndian) {
     }
 
     public static char readCharVolatile(byte[] buffer, int pos, boolean bigEndian) {
-        return readCharVolatile(buffer, (long) (BYTE_ARRAY_BASE_OFFSET + pos), bigEndian);
+        return readCharVolatile(BYTE_BUFFER_MEMORY_ACCESSOR, buffer, pos, bigEndian);
```

Comment:
Currently (without this PR), when `Unsafe` is enabled, we can read from `byte[]` as volatile by using `readXYZVolatile(byte[] buffer, int pos, boolean bigEndian)` signature by passing our `byte[]` object. 

But with this PR, since `BYTE_BUFFER_MEMORY_ACCESSOR` doesn't support volatile read, we must cast our `byte[]` instance to `Object` for calling `readXYZVolatile(Object base, long offset, boolean bigEndian)` explicitly to enforce using `UNSAFE_MEMORY_ACCESSOR`.

Same comment is also for volatile writes.